Allow small text in Markdown

jrieke opened the pull request:
## Describe your changes

Allows to add small text in all features that support Markdown via `:small[foo]`. E.g.:

```python
st.markdown("Normal text and :small[small text]")
```

This renders as:

![Screenshot 2025-02-22 at 02 05 16](https://github.com/user-attachments/assets/28ef38d1-a938-4248-a5cd-6b8b29fbd847)

The small text has the same size as `st.caption` and widget labels, i.e. 14px by default. It can be used in combination with all other Markdown features. 

## GitHub Issue Link (if applicable)

Closes #10486

## Testing Plan

Added a JS unit test for the feature and added it to one of our snapshot e2e tests for markdown features. 

---

**Contribution License Agreement**

By submitting this pull request you agree that all contributions to this project are made under the Apache 2.0 license.

jrieke committed:
Add small text detection in remarkColoring

jrieke committed:
Add JS unit tests

jrieke committed:
Remove test for combinations, use theme properly

jrieke committed:
Add small to e2e snapshot test

jrieke committed:
Update snapshots

sfc-gh-lwilby commented:
This seems to make small and colour exclusive -- i.e. you can't mark something with small and a colour? Or is that just for the inline marking?

jrieke committed:
Add e2e test for small red

jrieke commented:
> This seems to make small and colour exclusive -- i.e. you can't mark something with small and a colour? Or is that just for the inline marking?

Nope, you can do both. Added an example where it does `:small[:red[small red]]` to the e2e test. Each of these tags is treated as a different text directive, so it will apply both styles if you nest them. 

Also added docstrings now.

jrieke committed:
Update snapshots

jrieke committed:
Add docstring
